### PR TITLE
parser: fix some parse error for parenthese subqueries

### PR DESCRIPTION
**Pull request by lzmhhh123**
Signed-off-by: lzmhhh123 <lzmhhh123@gmail.com>

### What problem does this PR solve? 
close https://github.com/pingcap/parser/issues/1127

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change

Side effects

 - Possible performance regression
 - Increased code complexity

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Review by bb7133 — APPROVED** [review]

**Comment by bb7133** [comment]
Please resolve the conflicts~

File: parser.y [hunk 1]
```diff
@@ -4022,16 +4024,48 @@ CreateTableSelectOpt:
...
 	{
-		$$ = $2
+		var sel ast.StmtNode
+		switch x := $1.(*ast.SubqueryExpr).Query.(type) {
+		case *ast.SelectStmt:
+			x.IsInBraces = true
+			sel = x
+		case *ast.SetOprStmt:
+			x.IsInBraces = true
+			sel = x
```
Comment (xiongjiwei) [inline parser.y:4066]:
```suggestion
		case *ast.SelectStmt, *ast.SetOprStmt:
			x.IsInBraces = true
			sel = x
```

File: parser.y [hunk 2]
```diff
@@ -10674,20 +10743,29 @@ Statement:
...
+|	SubSelect
+	{
+		var sel ast.StmtNode
+		switch x := $1.(*ast.SubqueryExpr).Query.(type) {
+		case *ast.SelectStmt:
+			x.IsInBraces = true
+			sel = x
+		case *ast.SetOprStmt:
+			x.IsInBraces = true
+			sel = x
```
Comment (xiongjiwei) [inline parser.y:10758]: ditto

File: parser.y [hunk 3]
```diff
@@ -10715,7 +10808,22 @@ ExplainableStmt:
...
+		}
```
Comment (xiongjiwei) [inline parser.y:10824]: ditto

File: parser.y [hunk 4]
```diff
@@ -12187,7 +12295,22 @@ RoleSpecList:
...
+|	SubSelect
+	{
+		var sel ast.StmtNode
+		switch x := $1.(*ast.SubqueryExpr).Query.(type) {
+		case *ast.SelectStmt:
+			x.IsInBraces = true
+			sel = x
+		case *ast.SetOprStmt:
+			x.IsInBraces = true
+			sel = x
```
Comment (xiongjiwei) [inline parser.y:12310]: ditto

File: parser.y (hunk 1 above)
Comment (lzmhhh123) [reply parser.y]: No, golang can't do this. In this statement, x will be regarded as `ResultNode`.

Comment (lzmhhh123) [reply parser.y]: No, golang can't do this. In this statement, x will be regarded as `ResultNode`.

**Review by wjhuang2016 — APPROVED** [review]

**Pull request merged**